mlflow/pipelines/steps/predict.py

```diff
@@ -122,18 +131,27 @@ def _run(self, output_directory):
             scored_sdf.write.format("delta").saveAsTable(self.step_config["output_location"])
 
         # predict step artifacts
-        scored_sdf.coalesce(1).write.format("parquet").save(
-            os.path.join(output_directory, _SCORED_OUTPUT_FILE_NAME)
-        )
-
-        scored_size = scored_sdf.count()
-        if scored_size > _MAX_PROFILE_SIZE:
-            sample_percentage = _MAX_PROFILE_SIZE / scored_size
-            scored_sdf = scored_sdf.sample(sample_percentage)
+        if databricks_utils.is_in_databricks_runtime():
+            dbfs_path = os.path.join(
+                ".mlflow",
+                _get_execution_directory_basename(self.pipeline_root),
+                _SCORED_OUTPUT_FOLDER_NAME,
+            )
```

Comment:
In Databricks, it seems like the spark writes parquet files as a folder. This makes it difficult for the `get_artifact` logic to work because it reads the data path as if it were a file. Thus, I've updated the `scored_data` artifact writing and reading logic to a path that does not have a `.parquet` file extension.